File: back/src/test/java/com/tdevilleduc/urthehero/back/model/PageTest.java
package com.tdevilleduc.urthehero.back.model;

import com.tdevilleduc.urthehero.back.utils.TestUtils;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class PageTest {

    @Test
    void test_Constructor() {
        Integer pageId = 46;
        String pageText = "test de titre";
        String pageImage = "image de ouf";
        Story story = TestUtils.createRandomStory();
        Page page = new Page(pageId, pageText, pageImage);

        Assertions.assertEquals(page.getId(), pageId);
        Assertions.assertEquals(page.getText(), pageText);
        Assertions.assertEquals(page.getImage(), pageImage);
        Assertions.assertTrue(page.getNextPageList().isEmpty());

        Page secondPage = new Page();
        secondPage.setId(pageId);
        secondPage.setText(pageText);
        secondPage.setImage(pageImage);

        Assertions.assertEquals(secondPage.toString(), page.toString());
        Assertions.assertEquals(secondPage, page);
    }
}

File: back/src/test/java/com/tdevilleduc/urthehero/back/model/PageTest.kt
package com.tdevilleduc.urthehero.back.model

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class PageTest {
    @Test
    fun test_Constructor() {
        val pageId = 46
        val pageText = "test de titre"
        val pageImage = "image de ouf"
        val pageStoryId = 27
        val page = Page(pageId, pageText, pageImage)
        page.storyId = pageStoryId
        Assertions.assertEquals(page.id, pageId)
        Assertions.assertEquals(page.text, pageText)
        Assertions.assertEquals(page.image, pageImage)
        Assertions.assertEquals(page.storyId!!, pageStoryId)
        Assertions.assertTrue(page.nextPageList.isEmpty())
    }
}